tests/sync/partial_sync.cpp

```diff
@@ -287,20 +300,146 @@ TEST_CASE("Partial sync", "[sync]") {
     }
 
     SECTION("re-registering the same query with the same name on the same type should succeed") {
-        run_query("number > 1", partial_config, "object_a", "query"s, [](Results results, std::exception_ptr error) {
+        subscribe_and_wait("number > 1", partial_config, "object_a", "query"s, [](Results results, std::exception_ptr error) {
             REQUIRE(!error);
             REQUIRE(results.size() == 2);
             REQUIRE(results_contains(results, {2, 2, "partial"}));
             REQUIRE(results_contains(results, {3, 8, "sync"}));
         });
 
-        run_query("number > 1", partial_config, "object_a", "query"s, [](Results results, std::exception_ptr error) {
+        subscribe_and_wait("number > 1", partial_config, "object_a", "query"s, [](Results results, std::exception_ptr error) {
             REQUIRE(!error);
             REQUIRE(results.size() == 2);
             REQUIRE(results_contains(results, {2, 2, "partial"}));
             REQUIRE(results_contains(results, {3, 8, "sync"}));
         });
     }
+
+    SECTION("unnamed query can be unsubscribed while in creating state") {
+        auto subscription = subscription_with_query("number > 1", partial_config, "object_a", util::none);
+
+        std::atomic<bool> partial_sync_done(false);
```

Comment:
These shouldn't need to be atomic as both the callback and run_until() are called on the main thread.